extensions/ql-vscode/src/remote-queries/gh-actions-api-client.ts

```diff
@@ -154,13 +154,30 @@ async function listWorkflowRunArtifacts(
   workflowRunId: number
 ) {
   const octokit = await credentials.getOctokit();
-  const response = await octokit.rest.actions.listWorkflowRunArtifacts({
-    owner,
-    repo,
-    run_id: workflowRunId,
-  });
 
-  return response.data.artifacts;
+  // There are limits on the number of artifacts that are returned by the API
+  // so we use paging to make sure we retrieve all of them.
+  let morePages = true;
+  let pageNum = 1;
+  const allArtifacts = [];
+
+  if (morePages) {
```

Comment:
I don't think this currently goes through all the pages - should this be a `while` loop instead?

Reply:
Sorry I thought I wrote `while` there 😹

Reply:
Fixed